personal_finances/user/user_auth.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import json
+import logging
+import boto3
+import bcrypt
+import jwt
+import os
+from datetime import datetime, timedelta
+from typing import Any, Tuple
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AuthorizationNotFound(Exception):
+    def __str__(self) -> str:
+        return "AuthorizationNotFoundException"
+
+
+class UserNotFound(Exception):
+    def __str__(self) -> str:
+        return "UserNotFoundException"
+
+
+class UserInvalidPassword(Exception):
+    def __str__(self) -> str:
+        return "UserInvalidPasswordException"
+
+
+def _decode_basic_auth(auth_header: str) -> Tuple[str, str]:
+    try:
+        auth_value = auth_header.replace("Basic ", "")
+        decoded_bytes = base64.b64decode(auth_value)
+        decoded_str = decoded_bytes.decode("utf-8")
+        userId, password = decoded_str.split(":", 1)
+        return userId, password
+    except Exception:
+        raise AuthorizationNotFound()
+
+
+def _get_auth_header(event: dict) -> str:
+    auth_header = str(event.get("headers", {}).get("Authorization", ""))
+
+    if not auth_header:
+        raise AuthorizationNotFound()
+
+    return auth_header
+
+
+def _get_user(userId: str) -> dict[Any, Any]:
+    dynamodb = boto3.client("dynamodb")
+    response = dynamodb.get_item(
+        TableName=os.environ["INFINEAT_DYNAMODB_USER_TABLE_NAME"],
+        Key={"userId": {"S": userId}},
+    )
+
+    user: dict = response.get("Item")
+    if not user:
+        raise UserNotFound(f"User not found: {userId}")
+
+    return user
```

Comment:
Better not to leak the dynamodb format outside this function. I'd probably call this function `_get_stored_password` and make it return a plain string with the password value.